python/ray/data/_internal/logical/rules/operator_fusion.py

```diff
@@ -130,6 +131,9 @@ def _can_fuse(self, down_op: PhysicalOperator, up_op: PhysicalOperator) -> bool:
         down_logical_op = self._op_map[down_op]
         up_logical_op = self._op_map[up_op]
 
+        if isinstance(up_logical_op, Read) and not up_logical_op.fusable():
```

Comment:
Why the extra check if it's a `Read` op?

Reply:
The `fusable` method is part of the Read class only.

Reply:
I think we can define `fusable` in the base LogicalOperator class. Other op may need it as well in the future.